SAMPLE_FORMAT.md

```diff
@@ -0,0 +1,130 @@
+ allowed. A parameter for the type of a file or a specific action is not.
+ 
+Any declared function parameters should include a commented out example of how that parameter could
+ be declared. This provides both an example of how to derive the variables and what information they 
+ should represent. 
+ 
+### Function Return Type
+The function in the snippet should not return any value. Instead, they should print the results of
+ actions to the console to be validated later. 
+ 
+### Exception Handling
+The snippet should show how to correctly handle Exceptions that occur during the snippet. Top level 
+ exceptions can be handled by logging the exception to `System.out.println`. If the exception is 
+ something the user may commonly encounter, include a comment explaining how to avoid or handle 
+ correctly.
+
+Example:
+```java
+try {
+  // Do something
+} catch (Exception e) {
+  System.out.println("Error during functionName: \n" + e.toString());
+}
+```
+
+### Client Initialization
+The preferred style for initialization is to use a try-with-resources statement with a comment 
+clarifying how to handle multiple requests and clean up instructions. 
+
+Example:
+```java
+// Initialize client that will be used to send requests. This client only needs to be created
+// once, and can be reused for multiple requests. After completing all of your requests, call
+// the "close" method on the client to safely clean up any remaining background resources.
+try (DlpServiceClient dlp = DlpServiceClient.create()) {
+  // Do something
+} catch (Exception e) {
+  System.out.println("Error during functionName: \n" + e.toString());
+}
+```
+
+### Arrange, Act, Assert
+Samples should generally follow the following outline: 
+* _Arrange_ - Create and configure the components for the request. Avoid nesting these components,
+ as often they use Builders which can hinder readibility. 
+* _Act_ - Send the request and receive the response.
+* _Assert_ - Verify the call was successful or that the response is correct. This is often done by
+ print contents of the response to stdout. 
+
+### Testing
+Snippets should have tests that should verify the snippet works and compiles correctly. Creating 
+ mocks for these tests are optional. These tests should capture output created by the snippet to 
+ verify that it works correctly. See the tests in the cannoncial for an example of how to do this
+ correctly. 
+
+### External Resources
+Use of environment variables over system properties is strongly preferred for configuration. 
+
+Any additional files required should be stored in `src/test/resources`.
+
+## Additional Best Practices 
+
+The following are some general Java best practices that should be followed in samples to remain
+idiomatic. 
+
+### Time
+Use the `java.time` package when dealing with units of time in some manner. 
+
+### Logging
+Use `java.util.logging` for consistent logging in web applications. 
```

Comment:
Should we move to https://github.com/google/flogger at some point?

Reply:
Added to todo list for discussion.